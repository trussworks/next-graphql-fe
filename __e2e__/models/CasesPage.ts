import { Locator, Page } from "@playwright/test"

const apiURL = process.env.PLAYWRIGHT_URL

export class CasesPage {
  readonly page: Page
  readonly h1: Locator
  readonly paragraph: Locator
  readonly table: Locator
  readonly tableHead: Locator
  readonly tableHeaders: Locator
  readonly tableBody: Locator
  readonly tableRows: Locator

  constructor(page: Page) {
    this.page = page
    this.h1 = page.locator("h1")
    this.paragraph = page.locator('p[role="paragraph"]')
    this.table = page.locator("table")
    this.tableHead = this.table.locator("thead")
    this.tableHeaders = this.tableHead.locator("th")
    this.tableBody = this.table.locator("tbody")
    this.tableRows = this.tableBody.locator("tr")
  }

  async goto() {
    await this.page.goto(`${apiURL}/incidents`)
  }
}
