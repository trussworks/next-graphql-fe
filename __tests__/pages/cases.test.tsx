import React from "react"
import { screen } from "@testing-library/react"
import {
  renderWithQueryClient as render,
  validateA11y,
} from "../../utils/test-utils"
import CasesPage from "../../pages/cases"

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/cases",
      pathname: "/cases",
      query: {},
      asPath: "/cases",
    }
  },
}))

// TODO: Until we get MSW setup, we can only test the no-data state
describe("Cases Page", () => {
  it("displays page content while loading", () => {
    render(<CasesPage />)

    const h1 = screen.getByRole("heading")
    expect(h1).toBeInTheDocument()
    expect(h1.textContent).toMatchInlineSnapshot(`"All Alerts"`)

    const subHeading = screen.getByRole("paragraph")
    expect(subHeading).toBeInTheDocument()
    expect(subHeading.textContent).toMatchInlineSnapshot(
      `"View of all alerts that are active in the system today."`
    )

    const loadingText = screen.getByText(/loading.../i)
    expect(loadingText).toBeInTheDocument()
  })

  it("matches snapshot while", () => {
    const { container } = render(<CasesPage />)

    expect(container).toMatchSnapshot()
  })

  it("is accessiable while loading", async () => {
    const { container } = render(<CasesPage />)

    await validateA11y(container)
  })
})
