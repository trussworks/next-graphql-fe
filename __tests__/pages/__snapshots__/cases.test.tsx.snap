// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cases Page matches snapshot while loading 1`] = `
<div>
  <main
    class="cases"
  >
    <h1>
      All Alerts
    </h1>
    <p
      role="paragraph"
    >
      View of all alerts that are active in the system today.
    </p>
    loading...
  </main>
</div>
`;

exports[`Cases Page matches snapshot with fetched data 1`] = `
<div>
  <main
    class="cases"
  >
    <h1>
      All Alerts
    </h1>
    <p
      role="paragraph"
    >
      View of all alerts that are active in the system today.
    </p>
    <table
      class="usa-table usa-table--borderless src-components-Table-Table-module__fullwidth--fOyj6"
      data-testid="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Color Code
          </th>
          <th
            scope="col"
          >
            Name
          </th>
          <th
            scope="col"
          >
            Date received
          </th>
          <th
            scope="col"
          >
            Case status
          </th>
          <th
            scope="col"
          >
            InT Analyst Assigned
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="row"
        >
          <td>
            <span
              class="usa-tag gray"
              data-testid="tag"
            >
              GRAY
            </span>
          </td>
          <th
            scope="row"
          >
            Jason
             
            Ash
          </th>
          <td>
            12/10/2021
          </td>
          <td>
            PRE_INQUIRY
          </td>
          <td>
            <span
              class="usa-tag todo"
              data-testid="tag"
            >
              To do
            </span>
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            <span
              class="usa-tag gray"
              data-testid="tag"
            >
              GRAY
            </span>
          </td>
          <th
            scope="row"
          >
            Riley
             
            Baker
          </th>
          <td>
            12/10/2021
          </td>
          <td>
            PRE_INQUIRY
          </td>
          <td>
            Jonathan
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            <span
              class="usa-tag blue"
              data-testid="tag"
            >
              BLUE
            </span>
          </td>
          <th
            scope="row"
          >
            Skyler
             
            Hunt
          </th>
          <td>
            12/10/2021
          </td>
          <td>
            PRE_INQUIRY
          </td>
          <td>
            <span
              class="usa-tag todo"
              data-testid="tag"
            >
              To do
            </span>
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            <span
              class="usa-tag gray"
              data-testid="tag"
            >
              GRAY
            </span>
          </td>
          <th
            scope="row"
          >
            Susan
             
            Lee
          </th>
          <td>
            12/10/2021
          </td>
          <td>
            IN_PROGRESS
          </td>
          <td>
            <span
              class="usa-tag todo"
              data-testid="tag"
            >
              To do
            </span>
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            <span
              class="usa-tag gray"
              data-testid="tag"
            >
              GRAY
            </span>
          </td>
          <th
            scope="row"
          >
            Kerry
             
            Smith
          </th>
          <td>
            12/10/2021
          </td>
          <td>
            IN_PROGRESS
          </td>
          <td>
            <span
              class="usa-tag todo"
              data-testid="tag"
            >
              To do
            </span>
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            <span
              class="usa-tag blue"
              data-testid="tag"
            >
              BLUE
            </span>
          </td>
          <th
            scope="row"
          >
            Jessica
             
            Smith
          </th>
          <td>
            12/9/2021
          </td>
          <td>
            ZERO_INACTIVE
          </td>
          <td>
            Jonathan
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            <span
              class="usa-tag purple"
              data-testid="tag"
            >
              PURPLE
            </span>
          </td>
          <th
            scope="row"
          >
            Ted
             
            Smith
          </th>
          <td>
            12/9/2021
          </td>
          <td>
            ZERO_INACTIVE
          </td>
          <td>
            Lisa
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            <span
              class="usa-tag gray"
              data-testid="tag"
            >
              GRAY
            </span>
          </td>
          <th
            scope="row"
          >
            christopher
             
            swinglhurse-walters
          </th>
          <td>
            12/9/2021
          </td>
          <td>
            ZERO_INACTIVE
          </td>
          <td>
            Susan
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            <span
              class="usa-tag brown"
              data-testid="tag"
            >
              BROWN
            </span>
          </td>
          <th
            scope="row"
          >
            Susan
             
            Lee
          </th>
          <td>
            12/8/2021
          </td>
          <td>
            ZERO_INACTIVE
          </td>
          <td>
            Quinn
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            <span
              class="usa-tag brown"
              data-testid="tag"
            >
              BROWN
            </span>
          </td>
          <th
            scope="row"
          >
            Kerry
             
            Smith
          </th>
          <td>
            12/8/2021
          </td>
          <td>
            ZERO_INACTIVE
          </td>
          <td>
            Jo
          </td>
        </tr>
      </tbody>
    </table>
  </main>
</div>
`;
